manim/mobject/text/tex_mobject.py

```diff
@@ -381,6 +381,29 @@ def set_color_by_tex(self, tex, color, **kwargs):
             part.set_color(color)
         return self
 
+    def set_opacity_by_tex(
+        self, tex: str, opacity: float = 0.5, remaining_opacity: float = None, **kwargs
```

Comment:
Minor type hint mismatch

```suggestion
        self, tex: str, opacity: float = 0.5, remaining_opacity: float | None = None, **kwargs
```